.github/workflows/isolated_migrations.yml

```diff
@@ -0,0 +1,69 @@
+# GitHub action that checks:
+# - that the migrations and the related code changes are not in the same pull request
+# - that the schema.rb and data_schema.rb files are included in the pull request that introduces the corresponding migration
+# Based on https://github.com/marketplace/actions/changed-files
+
+name: Isolated Migrations
+
+on:
+  pull_request
+
+jobs:
+  check_changed_files:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Group files that have changed
+        id: changed-files-yaml
+        uses: tj-actions/changed-files@v39
+        with:
+          files_yaml: |
+            migrations:
+              - src/api/db/**
+            not_migrations:
+              - '!src/api/db/**'
+            schema_migrations:
+              - src/api/db/migrate/**
+            data_migrations:
+              - src/api/db/data/**
+            schema:
+              - src/api/db/schema.rb
+            data_schema:
+              - src/api/db/data_schema.rb
+
+      - name: List the files related to migrations that this PR changes
+        if: steps.changed-files-yaml.outputs.migrations_any_changed == 'true'
+        run: |
+          echo "Related to migrations:"
```

Comment:
This `echo` and the others in the steps below doesn't provide value when looking at this in the GitHub UI. With the step name, it's already clear what the listed files are. See this example below:

![ui](https://github.com/openSUSE/open-build-service/assets/1102934/a2c86cb0-79ec-46dc-bef0-4a0e75588859)

Reply:
Yes, we can omit them.